### ISSUE TITLE
[flake8-django] DJ001 is not raised for fields which has unique=True

### ISSUE THREAD

**Issue by kiddten** [astral-sh/ruff#13887]
```
class Product(models.Model):
    test = models.CharField(null=True, blank=True, max_length=255)
```
is correct DJ001 Avoid using `null=True` on string-based fields such as `CharField`

but

```
class Product(models.Model):
    test = models.CharField(null=True, blank=True, max_length=255, unique=True)
```

> All checks passed!

gives no warning

ruff version 0.7.0

**Comment by kiddten** [comment]
Hmm looks like this is expected behaviour https://github.com/astral-sh/ruff/blob/4d109514d608b45bc11a5ecf3e022112461cb571/crates/ruff_linter/src/rules/flake8_django/rules/nullable_model_string_field.rs#L105

**Comment by UnknownPlatypus** [comment]
See https://docs.djangoproject.com/en/5.1/ref/models/fields/#null for the rationale. This is indeed expected behavior
